gradoop-common/src/main/java/org/gradoop/common/model/impl/properties/PropertyValue.java

```diff
@@ -172,7 +172,7 @@ public PropertyValue() { }
    * @param value value with supported type
    */
   private PropertyValue(Object value) {
-    setObject(value);
+    setObject(value instanceof PropertyValue ? ((PropertyValue) value).getObject() : value);
```

Comment:
Ah - is used to copy the property value. As an alternative, a `copy()` method can be added. Maybe more meaningful.

Reply:
I think `copy()` would be good. Any other opinions?

Reply:
You can implement the  `clone()`  method and the `Cloneable` interface

Reply:
Maybe implementing the Cloneable blows up the class. Since the PropVal is used in many objects, its possibly not the best option. See https://dzone.com/articles/java-cloning-copy-constructor-vs-cloning

Reply:
Since LogicalGraph also has a `copy()` method, i will do it this way.